lib/lambda_ethereum_consensus/fork_choice/helpers.ex

```diff
@@ -2,12 +2,35 @@ defmodule LambdaEthereumConsensus.ForkChoice.Helpers do
   @moduledoc """
     Utility functions for the fork choice.
   """
+  alias LambdaEthereumConsensus.ForkChoice
   alias LambdaEthereumConsensus.StateTransition.{Accessors, Misc}
+  alias LambdaEthereumConsensus.Store.{BlockStore, StateStore}
   alias SszTypes.BeaconBlock
   alias SszTypes.BeaconState
   alias SszTypes.Checkpoint
   alias SszTypes.Store
 
+  @spec current_status_message(Store.t()) ::
+          {:ok, SszTypes.StatusMessage.t()} | {:error, any}
+  def current_status_message(store) do
+    with {:ok, head_root} <- get_head(store),
+         {:ok, state} <- StateStore.get_state(head_root),
+         {:ok, signed_head_block} <-
+           BlockStore.get_block(head_root),
+         {:ok, finalized_checkpoint} <-
+           ForkChoice.Store.get_finalized_checkpoint() do
+      {:ok,
+       %SszTypes.StatusMessage{
+         fork_digest:
+           Misc.compute_fork_digest(state.fork.current_version, state.genesis_validators_root),
+         finalized_root: finalized_checkpoint.root,
+         finalized_epoch: finalized_checkpoint.epoch,
+         head_root: head_root,
+         head_slot: signed_head_block.message.slot
```

Comment:
it seems you are getting the slot from the block, but you can also get it from the state, so no need to retrieve the block.